frontend/packages/local-storage-operator-plugin/src/components/local-volume-set/create-local-volume-set.tsx

```diff
@@ -0,0 +1,236 @@
+
+  const { ns, appName } = match.params;
+  const modelName = LocalVolumeSetModel.label;
+
+  const toggleShowNodesList = () => {
+    setShowNodesList(!showNodesList);
+  };
+
+  const onSubmit = (event: React.FormEvent<EventTarget>) => {
+    event.preventDefault();
+
+    const requestData: LocalVolumeSetKind = {
+      apiVersion: LocalVolumeSetModel.apiVersion,
+      kind: LocalVolumeSetModel.kind,
+      metadata: { name: volumeSetName },
+      spec: {
+        storageClassName,
+        volumeMode,
+        deviceInclusionSpec: {
+          // Only Raw disk supported for 4.5
+          deviceTypes: [DeviceType.RawDisk],
+          deviceMechanicalProperty: [DeviceMechanicalProperty[volumeType]],
+        },
+      },
+    };
+
+    if (showNodesList) {
+      const selectedNodesUID = getSelectedNodeUIDs(rows);
+      const selectedNodes = selectedNodesUID.map((uid) => rows[uid].props.data.metadata.name);
+      requestData.spec.nodeSelector = {
+        nodeSelectorTerms: [
+          {
+            matchExpressions: [
+              { key: 'kubernetes.io/hostname', operator: 'In', values: [...selectedNodes] },
+            ],
+          },
+        ],
+      };
+    }
+    if (maxVolumeLimit) requestData.spec.maxDeviceCount = +maxVolumeLimit;
+
+    handlePromise(k8sCreate(LocalVolumeSetModel, requestData))
+      .then((resource) => history.push(resourceObjPath(resource, referenceFor(resource))))
+      .catch(() => null);
+  };
+
+  return (
+    <>
+      <div className="co-create-operand__header">
+        <div className="co-create-operand__header-buttons">
+          <BreadCrumbs
+            breadcrumbs={[
+              {
+                name: 'Local Storage',
+                path: resourcePathFromModel(ClusterServiceVersionModel, appName, ns),
+              },
+              { name: `Create ${modelName}`, path: '' },
+            ]}
+          />
+        </div>
+        <h1 className="co-create-operand__header-text">{`Create ${modelName}`}</h1>
+        <p className="help-block">
+          A {modelName} allows you to filter a set of storage volumes, group them and create a
+          dedicated storage class to consume storage for them.
+        </p>
+      </div>
+      <Form noValidate={false} className="co-m-pane__body co-m-pane__form" onSubmit={onSubmit}>
+        <FormGroup label="Volume Set Name" isRequired fieldId="create-lvs--volume-set-name">
+          <TextInput
+            type={TextInputTypes.text}
+            id="create-lvs--volume-set-name"
+            value={volumeSetName}
+            onChange={setVolumeSetName}
+            isRequired
+          />
+        </FormGroup>
+        <FormGroup label="Storage Class Name" fieldId="create-lvs--storage-class-name">
+          <TextInput
+            type={TextInputTypes.text}
+            id="create-lvs--storage-class-name"
+            value={storageClassName}
+            onChange={setStorageClassName}
```

Comment:
This would show some errors in the console. `onChange` send two params whereas `setStorageClass` only accepts one.

Reply:
Sure :+1: